pkg/server/v3/delta_test.go

```diff
@@ -0,0 +1,413 @@
+package server_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	rsrc "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+)
+
+func (config *mockConfigWatcher) CreateDeltaWatch(req *discovery.DeltaDiscoveryRequest, st *stream.StreamState) (chan cache.DeltaResponse, func()) {
+	config.deltaCounts[req.TypeUrl] = config.deltaCounts[req.TypeUrl] + 1
+
+	// Create our out watch channel to return with a buffer of one
+	out := make(chan cache.DeltaResponse, 1)
+
+	if len(config.deltaResponses[req.TypeUrl]) > 0 {
+		res := config.deltaResponses[req.TypeUrl][0]
+		// In subscribed, we only want to send back what's changed if we detect changes
+		var subscribed []types.Resource
```

Comment:
Since we're in test world, how about instead of filtering the responses we instead verify that the response matches resources is a superset of the resources that have been requested for this stream?

Reply:
We could do that, if that is going to be inside CreateDeltaWatch though, how exactly would that get used elsewhere in the mock stream tests? If we just verify the superset, I don't think that's enough to have a functioning test. I think we also need to continue passing down filtered responses with also checking the superset? If that makes any sense. Let me know what you think!